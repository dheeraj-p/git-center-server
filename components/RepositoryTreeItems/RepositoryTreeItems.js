import Link from 'next/link';
import FinderIcon from '../FinderIcon/FinderIcon';

export default function RepositoryTreeItems({ repository, branch, content }) {
  return content.map(({ path, type, name }) => {
    return (
      <Link
        href={`/repository/[...path]`}
        as={`/repository/${repository}/${type}/${branch}/${path}`}
      >
        <a className='list-group-item list-group-item-action'>
          <FinderIcon type={type} />
          <span className='text-muted'>{name}</span>
        </a>
      </Link>
    );
  });
}
