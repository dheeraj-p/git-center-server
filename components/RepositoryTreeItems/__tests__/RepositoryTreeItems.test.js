import { shallow } from 'enzyme';
import RepositoryTreeItems from '../RepositoryTreeItems';

describe('RepositoryTreeItems', () => {
  let content;
  beforeEach(() => {
    content = [
      {
        type: 'blob',
        name: 'testBlob',
        path: 'src/blob.js',
      },
      {
        type: 'tree',
        name: 'src',
        path: '/src',
      },
    ];
  });

  it('should render properly', () => {
    const wrapper = shallow(
      <RepositoryTreeItems
        content={content}
        repository='testRepo'
        branch='master'
      />
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
