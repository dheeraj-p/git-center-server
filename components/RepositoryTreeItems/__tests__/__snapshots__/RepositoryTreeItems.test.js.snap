// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryTreeItems should render properly 1`] = `
"<Link href=\\"/repository/[...path]\\" as=\\"/repository/testRepo/blob/master/src/blob.js\\">
  <a className=\\"list-group-item list-group-item-action\\">
    <FinderIcon type=\\"blob\\" />
    <span className=\\"text-muted\\">
      testBlob
    </span>
  </a>
</Link>


<Link href=\\"/repository/[...path]\\" as=\\"/repository/testRepo/tree/master//src\\">
  <a className=\\"list-group-item list-group-item-action\\">
    <FinderIcon type=\\"tree\\" />
    <span className=\\"text-muted\\">
      src
    </span>
  </a>
</Link>"
`;
